packages/shared/src/utils/logger.ts

```diff
@@ -28,10 +28,19 @@ type LoggerContext = {
  * @returns The default logging level.
  */
 function getDefaultLevel(): LogLevel {
+  if (
+    // eslint-disable-next-line no-restricted-globals
+    process?.env?.NODE_ENV === 'production' &&
+    // eslint-disable-next-line no-restricted-globals
+    process?.env?.ENABLE_LOGGING !== 'true'
+  ) {
+    return LogLevel.ERROR + 1;
```

Comment:
Maybe we could add a dedicated `LogLevel.SILENT/OFF` rather than using `LogLevel.ERROR + 1` for clarity.